tensorboard/plugins/scalar/tf_scalar_dashboard/tf-scalar-dashboard.html

```diff
@@ -286,9 +288,67 @@
         });
       },
 
-      _makeCategories(runToTag, selectedRuns, tagFilter) {
+      _makeCategories(runToTagInfo, selectedRuns, tagFilter) {
+        const runToTag = _.mapValues(runToTagInfo, x => Object.keys(x));
         return categorizeTags(runToTag, selectedRuns, tagFilter);
       },
+      _tagMetadata(runToTagInfo, runs, tag) {
+        /*
+         * Each run--tag combination may have its own display name and
+         * description, but we display just one chart per tag: this is a
+         * dimension mismatch. We reconcile this as follows:
+         *
+         *   - We only show a display name if all runs agree. Otherwise,
+         *     we use the tag name without the "/scalar_summary" suffix.
+         *
+         *   - If all runs agree on a description, we use it. Otherwise,
+         *     we concatenate all descriptions, annotating which ones
+         *     came from which run, and display them in a list.
+         */
+        let unanimousDisplayName = undefined;
+        const descriptions = {};  // map from description to runs it appears in
+        runs.forEach(run => {
+          const info = runToTagInfo[run][tag];
+          if (unanimousDisplayName === undefined) {
+            unanimousDisplayName = info.displayName;
+          }
+          if (unanimousDisplayName !== info.displayName) {
+            unanimousDisplayName = null;
+          }
+          if (descriptions[info.description] === undefined) {
+            descriptions[info.description] = [];
+          }
+          descriptions[info.description].push(run);
+        });
+        const displayName =
+          unanimousDisplayName !== null ?
+          unanimousDisplayName :
+          tag.replace(/\/scalar_summary$/, '');
+        const description = (() => {
+          if (Object.keys(descriptions).length === 1) {
+            return Object.keys(descriptions)[0];
+          } else {
+            const items = Object.keys(descriptions).map(description => {
+              const runs = descriptions[description].map(run => {
+                const safeRun = run
```

Comment:
This is thought-out, albeit later, we may want to generalize this code or move it somewhere so that all plugins can use it.